### DISCUSSION TITLE
Prevent `generateMetadata()` from destroying TTFB metric

### DISCUSSION THREAD

**Discussion by l-you** [vercel/next.js#50809] [App Router] [answered]
We use `async generateMetadata()` in our project on some critical frequently visited pages.
Speed of `fetch()` request used inside `generateMetadata` is slow. It cannot be faster.
Cache would help in this situation, but it does not work. This destroys our TTFB metric and initial page load time.
We use Vercel for production.
Both `revalidate` property for `fetch()` and `revalidate` property exported from `page.tsx` does not help. 
Also, our api provides `Cache-Control` response header, but NextJS does not save that cache locally for future requests.

Any suggestion on how to cache the result of `generateMetadata` for a 3 hours?

Example of our use case:
```typescript
export const staticHttpGet = async <D extends HTTPGetInterface>(
	url: D['url'],
	params: D['params'],
	next: NextFetchRequestConfig | undefined = undefined
): Promise<D['result']> => {
	let path = `https://some_url.com${url}`
	if (params !== undefined) {
		path += '?' + toHTTPQueryString(params)
	}
	return fetch(path, {
		method: 'GET',
		headers: {
			Accept: 'application/json',
		},
		next,
	})
		.then(resp => resp.json())
		.then(handleJson)
}

const queryFilters = async (params: GetAvailableFilters['params']) => {
	return await staticHttpGet<GetAvailableFilters>(
		'/api/available-product-filters',
		params,
		{
			revalidate: 600,
		}
	)
}
const queryProducts = async (
	productParams: SearchProductFrontEnd['params']
) => {
	return await staticHttpGet<SearchProductFrontEnd>(
		'/api/products',
		productParams,
		{
			revalidate: 600,
		}
	)
}
export const generateMetadata = async ({
	params,
}: {
	params: Params
}): Promise<Metadata> => {
	const [filters, products] = await Promise.all([
		queryFilters(getAvailableFilterParams(params)),
		queryProducts({
			...getDefaultParams(params),
			takeCount: 1,
			page: 1,
		}),
	])

	return {
                title:'....'
		//some stuff based on 'filters' and 'products' variable
	}
}
```

**Comment by l-you** [comment, marked as answer]
It seems that 'appDir' still has significant issues as of summer 2023. I hope it will be ready for production by the end of autumn.

**Reply by jetaggart** [reply]
@rusted-love did you figure out a solution to this? We're facing the same issues.

**Reply by l-you** [reply]
> @rusted-love did you figure out a solution to this? We're facing the same issues.

No. There are even more bugs than before. Next.js team is trying to fix them, but one bug fix produces even more bugs.

This issue can be related to current problem.
https://github.com/vercel/next.js/issues/43548
It was fixed, then broken, and then fixed again, and broken again.

**Reply by jetaggart** [reply]
Here's a larger tracking issue for this: https://github.com/vercel/next.js/issues/55524, might be helpful to add your experience in there.

**Comment by popovidis** [comment]
Any update on this? Running into this on one of the more recent builds.
